packages/server/src/auth/CrossmintAuth.ts

```diff
@@ -20,4 +30,62 @@ export class CrossmintAuth {
     public static from(crossmint: Crossmint): CrossmintAuth {
         return new CrossmintAuth(crossmint);
     }
+
+    public getJWKSUri() {
+        return `${this.apiClient.baseUrl}/.well-known/jwks.json`;
+    }
+
+    public verifyCrossmintJwtToken(token: string) {
+        return verifyCrossmintJwtToken(token, this.getJWKSUri());
+    }
+
+    private async refreshAuthMaterial(refreshToken: string): Promise<AuthMaterial> {
+        const result = await this.apiClient.post("/session/sdk/auth/refresh", {
+            body: JSON.stringify({ refresh: refreshToken }),
+        });
+
+        const resultJson = (await result.json()) as AuthMaterialResponse;
+
+        return {
+            jwtToken: resultJson.jwt,
+            refreshToken: resultJson.refresh,
+            user: resultJson.user,
+        };
+    }
+
+    public async getSession(options: GenericRequest | AuthMaterialBasic): Promise<AuthSession> {
+        const { jwtToken, refreshToken } =
+            "refreshToken" in options ? options : getAuthCookies(options as GenericRequest);
```

Comment:
I thought that TS would be able to infer that, if `"refreshToken"` is not there, then it is not of type `AuthMaterialBasic` so it must be `GenericRequest`. Does it still throw compiler errors if we remove the typecast?
```suggestion
            "refreshToken" in options ? options : getAuthCookies(options);
```